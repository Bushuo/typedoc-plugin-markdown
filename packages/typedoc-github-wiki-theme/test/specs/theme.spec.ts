import * as fs from 'fs';

import { TestApp } from 'typedoc-plugin-markdown/test/test-app';

let testApp: TestApp;

describe(`Theme:`, () => {
  describe(`url`, () => {
    testApp = new TestApp(['theme.ts']);
    test(`should getUrls'`, async () => {
      await testApp.bootstrap({ theme: './dist' });

      const urlMappings = testApp.theme.getUrls(testApp.project);
      expect(TestApp.getExpectedUrls(urlMappings)).toMatchSnapshot();
    });
  });
  describe(`sidebar`, () => {
    test(`should write sidebar for exports'`, async () => {
      testApp = new TestApp(['theme.ts']);
      await testApp.bootstrap({ theme: './dist', entryDocument: 'Home.md' });
      const sidebarFile = fs.readFileSync(testApp.tmpobj.name + '/_Sidebar.md');
      expect(sidebarFile.toString()).toMatchSnapshot();
    });
    test(`should write sidebar for modules'`, async () => {
      testApp = new TestApp(['breadcrumbs.ts', 'theme.ts']);
      await testApp.bootstrap({
        theme: './dist',
        entryDocument: 'Home.md',
        readme: 'none',
      });
      const sidebarFile = fs.readFileSync(testApp.tmpobj.name + '/_Sidebar.md');
      expect(sidebarFile.toString()).toMatchSnapshot();
    });
  });
});
