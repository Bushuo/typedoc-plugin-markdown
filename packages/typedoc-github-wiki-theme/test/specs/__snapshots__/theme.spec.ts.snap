// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Theme: sidebar should write sidebar for exports' 1`] = `
"## typedoc-plugin-markdown

- [Home](../wiki/Home)
- [Exports](../wiki/Exports)

### Namespaces

- [namespaceA](../wiki/Namespace:%20namespaceA)
- [namespaceB](../wiki/Namespace:%20namespaceB)

### Enumerations

- [enumItemA](../wiki/Enumeration:%20enumItemA)
- [enumItemB](../wiki/Enumeration:%20enumItemB)

### Classes

- [ClassItemA](../wiki/Class:%20ClassItemA)
- [ClassItemB](../wiki/Class:%20ClassItemB)
- [SomeNestedClass](../wiki/Class:%20SomeNestedClass)

### Interfaces

- [InterfaceItemA](../wiki/Interface:%20InterfaceItemA)
- [InterfaceItemB](../wiki/Interface:%20InterfaceItemB)
"
`;

exports[`Theme: sidebar should write sidebar for modules' 1`] = `
"## typedoc-plugin-markdown


### Modules

- [breadcrumbs](../wiki/Module:%20breadcrumbs)
- [theme](../wiki/Module:%20theme)

### Namespaces

- [namespaceA](../wiki/Namespace:%20namespaceA)
- [namespaceB](../wiki/Namespace:%20namespaceB)

### Enumerations

- [enumItemA](../wiki/Enumeration:%20enumItemA)
- [enumItemB](../wiki/Enumeration:%20enumItemB)

### Classes

- [Breadcrumbs](../wiki/Class:%20Breadcrumbs)
- [ClassItemA](../wiki/Class:%20ClassItemA)
- [ClassItemB](../wiki/Class:%20ClassItemB)
- [SomeNestedClass](../wiki/Class:%20SomeNestedClass)

### Interfaces

- [InterfaceItemA](../wiki/Interface:%20InterfaceItemA)
- [InterfaceItemB](../wiki/Interface:%20InterfaceItemB)
"
`;

exports[`Theme: url should getUrls' 1`] = `
Array [
  "Exports.md",
  "Exports.md#declarationitema",
  "Exports.md#declarationitemb",
  "Exports.md#functionitema",
  "Exports.md#functionitemb",
  "Home.md",
  "Exports.md#declarationitema",
  "Exports.md#declarationitemb",
  "Exports.md#functionitema",
  "Exports.md#functionitemb",
  "Namespace: namespaceA.md",
  "Class: SomeNestedClass.md",
  "Class: SomeNestedClass.md#constructor",
  "Namespace: namespaceB.md",
  "Enumeration: enumItemA.md",
  "Enumeration: enumItemB.md",
  "Class: ClassItemA.md",
  "Class: ClassItemA.md#constructor",
  "Class: ClassItemB.md",
  "Class: ClassItemB.md#constructor",
  "Interface: InterfaceItemA.md",
  "Interface: InterfaceItemA.md#prop",
  "Interface: InterfaceItemB.md",
  "Interface: InterfaceItemB.md#prop",
]
`;
